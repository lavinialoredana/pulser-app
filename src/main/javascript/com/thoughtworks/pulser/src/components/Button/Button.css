.Button{
    margin: 5rem;
    background: #89CFF0;
    font-size: 1em;
    font-weight:900;
    padding: 15px 20px;
    border: 1px solid #ddd;
}