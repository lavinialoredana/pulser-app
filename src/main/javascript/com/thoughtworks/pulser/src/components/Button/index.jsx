import React from 'react'
import "./Button.css";

const Button = ({ onClickButton, buttonName, isDisabled }) => {
  return (
    <div>
      <button 
        className="Button"
        data-testid="button"
        onClick={onClickButton}
        name={buttonName}
        disabled={isDisabled()}
      >
        {buttonName}
      </button>
    </div>
  );
};

export default Button