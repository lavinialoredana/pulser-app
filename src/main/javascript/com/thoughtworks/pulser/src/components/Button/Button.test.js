import { cleanup, render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Button from "./index.jsx";

describe("Button Component", () => {
  afterEach(() => {
    cleanup();
  });

  it("should match snapshot", ()=>{
    const {asFragment} = render(
        <Button/>
    )

    expect(asFragment()).toMatchSnapshot();
  })

it("should contain button", () => {
    render(<Button />);

    const buttonElement = screen.getByTestId("button");

    expect(buttonElement).toBeInTheDocument();
  });

  it("should fire click correctly", async ()=>{
    const user = userEvent.setup();
    const onClickMock =jest.fn();

    render(<Button onClickButton={onClickMock}/>);

    const button = screen.getByTestId("button");

    await user.click(button);
    expect(onClickMock).toHaveBeenCalled();

  })
});
