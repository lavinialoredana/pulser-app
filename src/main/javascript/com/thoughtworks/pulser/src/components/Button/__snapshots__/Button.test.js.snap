// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <button
      class="Button"
      data-testid="button"
      disabled=""
    />
  </div>
</DocumentFragment>
`;
